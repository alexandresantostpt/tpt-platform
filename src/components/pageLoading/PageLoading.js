import React, { useEffect } from 'react'
import styled from 'styled-components'

import { Power3, TweenMax } from 'gsap'

const StyledSpinner = styled.div`
    align-items: center;
    display: flex;
    height: 100vh;
    justify-content: center;
    width: 100vw;
`

const StyledSpinnerBall = styled.span`
    background: ${({ theme }) => theme.primary};
    border-radius: 100%;
    display: block;
    height: 20px;
    margin: 0 0.5rem;
    opacity: 0;
    width: 20px;
`

const PageLoading = () => {
    useEffect(() => {
        const balls = document.querySelectorAll('.js-spinnerBall')
        TweenMax.staggerFromTo(
            balls,
            0.75,
            {
                opacity: 0,
                scale: 0.1
            },
            {
                ease: Power3.easeInOut,
                opacity: 1,
                repeat: -1,
                scale: 1.2,
                yoyo: true
            },
            0.1
        )
    }, [])

    return (
        <StyledSpinner className="spinner">
            <StyledSpinnerBall className="spiiner__ball js-spinnerBall" />
            <StyledSpinnerBall className="spiiner__ball js-spinnerBall" />
            <StyledSpinnerBall className="spiiner__ball js-spinnerBall" />
            <StyledSpinnerBall className="spiiner__ball js-spinnerBall" />
            <StyledSpinnerBall className="spiiner__ball js-spinnerBall" />
        </StyledSpinner>
    )
}

export default PageLoading
