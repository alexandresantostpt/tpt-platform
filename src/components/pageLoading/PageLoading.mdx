---
name: PageLoading
route: /page/loading
---
import { Playground, PropsTable } from 'docz'

import PageLoading from './PageLoading'

import '@cfg/scripts'
import '@cfg/style'

# PageLoading

<PropsTable of={ PageLoading }/>

## Basic usage

```
<PageLoading/>
```

<Playground>
    <PageLoading/>
</Playground>
