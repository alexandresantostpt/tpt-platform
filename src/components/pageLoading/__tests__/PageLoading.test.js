import React from 'react'
import renderer from 'react-test-renderer'

import PageLoading from '../PageLoading'

describe('Tests for PageLoading component', () => {
    it('Should rendering without crash ', () => {
        const tree = renderer.create(<PageLoading />).toJSON()
        expect(tree).toMatchSnapshot()
    })
})
