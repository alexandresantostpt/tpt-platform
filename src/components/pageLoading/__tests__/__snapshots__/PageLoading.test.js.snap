// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests for PageLoading component Should rendering without crash  1`] = `
<div
  className="spinner PageLoading__StyledSpinner-wjb6ia-0 kXspCY"
>
  <span
    className="spiiner__ball js-spinnerBall PageLoading__StyledSpinnerBall-wjb6ia-1 lpcRdC"
  />
  <span
    className="spiiner__ball js-spinnerBall PageLoading__StyledSpinnerBall-wjb6ia-1 lpcRdC"
  />
  <span
    className="spiiner__ball js-spinnerBall PageLoading__StyledSpinnerBall-wjb6ia-1 lpcRdC"
  />
  <span
    className="spiiner__ball js-spinnerBall PageLoading__StyledSpinnerBall-wjb6ia-1 lpcRdC"
  />
  <span
    className="spiiner__ball js-spinnerBall PageLoading__StyledSpinnerBall-wjb6ia-1 lpcRdC"
  />
</div>
`;
